src/modules/queue/QueueComponent.jsx

```diff
@@ -21,79 +18,61 @@ class QueueContainer extends React.Component {
 	};
 
 	render() {
-		var queue = {};
-		if (this.state.queueOpen) {
-			queue.open = ' open';
-		} else {
-			queue.open = '';
+		if (!this.state.queueOpen) {
+			return (
+				<Positioning>
+					<QueueButton onClick={this.toggleQueue}>
+						<img src={queue_img} />
+					</QueueButton>
+				</Positioning>
+			);
 		}
 		return (
-			<QueueHolder className={queue.open}>
-				<div className="queue-list">
+			<Positioning>
+				<QueueList>
 					<QueueItem>
 						<div className="item-image" />
 						<div className="item-info">
 							<h4>Song Title</h4>
 						</div>
 					</QueueItem>
-				</div>
-				<QueueButton onClick={this.toggleQueue}>
-					<img src={queue_img} />
-					<div className="queue-x" />
-				</QueueButton>
-			</QueueHolder>
+					<QueueButton onClick={this.toggleQueue} className="open">
+						<img src={x_img} />
+					</QueueButton>
+				</QueueList>
+			</Positioning>
 		);
 	}
 }
 
-const QueueHolder = styled.div`
-	display: inline-block;
+const Positioning = styled.div`
 	position: absolute;
```

Comment:
Try to have as few `position: absolute` components as possible, particularly as part of the same component. They very quickly get complicated. Here it's `absolute`ly necessary (sorry), but we want to think carefully about making any of the children `absolute`.